sidecar/internal/server/server.go

```diff
@@ -53,7 +53,7 @@ func NewSidecarServer(endpoint string) *SidecarServer {
 	}
 
 	ss.scheme = "tcp"
-	ss.endpoint = ":" + endpoint
+	ss.endpoint = endpoint
```

Comment:
why can't get the podIP env and use it here?

Reply:
> why can't get the podIP env and use it here?

It can be done but I thought 
- It will be better to be explicit that `<pod_ip>:<port>` needs to passed and grpc server will be started to listen this address.
- prevent fetching of podIP env twice.

Reply:
>prevent fetching of podIP env twice.
 
fetch once and use it as it's a  single service.

anything is fine. I have seen in most of the places it is passed as the env. you also need to make sure the endpoint contains IP:port format

Reply:
some validation of the parameters would be good, do that immediately after `flag.Parse()`.

Reply:
added validations, please take a look